[Feature/lecture bookmark view] ✨ feat: bookmark 기능 추가

209512 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: REQ-LECT-005, 006
- 작업 요약: 북마크 api 목데이터 제거, 실제 기능 추가

## 📄 상세 내용
- [x] view 내 permission AllowAny -> IsAuthentication 변경
- [x] pagination class 추가
- [x] bookmark list get/create, bookmark delete 목데이터 제거
- [x] test_bookmark_integration.py 테스트코드 변경

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

209512 committed:
✨ feat(lecture): bookmark view 목데이터 삭제, 실제 로직 추가

pagination class 추가
lecture bookmark list  get/create, delete 목데이터 제거, 로직 추가.

209512 committed:
✅ test(lecture): bookmark 통합 테스트 변경

Meoyoug commented on apps/lecture/views/bookmark_views.py:
```diff
@@ -1,139 +1,140 @@
+                "data": {"count": queryset.count(), "next": None, "previous": None, "results": serializer.data},
+            },
+            status=status.HTTP_200_OK,
+        )
```
pagination에 필요한 쿼리파라미터가 없는경우 첫번째 페이지를 디폴트로 주는게 어떨까요

209512 committed:
♻️ refactor(lecture): 북마크 뷰 코드 간결화 및 예외 처리 개선

- GET 메서드 페이지네이션 미적용 분기 제거 (항상 페이지네이션 적용)
- request.query_params를 request.GET으로 변경하고 strip() 추가
- 페이지네이션 주석 추가
- POST 메서드 일반 예외 처리 제거 (전역 핸들러 활용)
- serializer_class 클래스 속성 추가

209512 committed:
✅ test(lecture): 북마크 통합테스트 변경(BaseLectureTest 상속)
